client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/site-preview/index.tsx

```diff
@@ -1,19 +1,43 @@
+import { SiteThumbnail, Spinner } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
-import React, { FC } from 'react';
+import { FC, useEffect, useRef } from 'react';
 import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
+import { useSitePreviewMShotImageHandler } from './hooks/use-site-preview-mshot-image-handler';
 import './style.scss';
 
 export const SitePreview: FC = () => {
 	const translate = useTranslate();
 	const fromUrl = useQuery().get( 'from' ) || '';
 
+	const previewRef = useRef< HTMLDivElement >( null );
+	const { mShotsOption, updateDimensions } = useSitePreviewMShotImageHandler();
+
+	useEffect( () => {
```

Comment:
Do you think this logic could be part of the `useSitePreviewMShotImageHandler` hook? I'm thinking this because it's only necessary for that logic, right?

Reply:
Thanks! I've moved it here https://github.com/Automattic/wp-calypso/pull/92355/commits/2c43bb4158b9359ca0b0b547018f4bc450614d13